src/filter.js

```diff
@@ -252,7 +270,7 @@ export class Filter extends CriteriaBuilder {
     }
   }
 
-  generateFields(columns, entityName) {
+  generateFields(columns, entityName, metaData) {
```

Comment:
Since `entityName` is optional, it might be nice to accept an `Object` instead.

For example change the method signature to:
```js
generateFields(columns, {entityName, metaData}={})
```

...so you can call it like this:
```js
this.generateFields(columns, {metaData})
```